IndentationCheck false positive with double-brace initialization

I have read check documentation: https://checkstyle.org/checks/misc/indentation.html  
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run  
I have executed the CLI and showed it below, as CLI describes the problem better than 1,000 words.

```bash
/var/tmp $ javac HashMapTestDefault.java
Successfully compiled with JDK 21

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>

/var/tmp $ cat HashMapTestDefault.java
import java.util.HashMap;

public class HashMapTestDefault {
    @Test
    void test() {
        var map = new HashMap<>() {{
            put("KEY1", "VALUE1");
            put("KEY2", "VALUE2");
        }};
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.3.0-all.jar -c config.xml HashMapTestDefault.java
Starting audit...
[ERROR] /var/tmp/HashMapTestDefault.java:7:13: 'block' child has incorrect indentation level 12, expected level should be one of the following: 16, 20. [Indentation]
[ERROR] /var/tmp/HashMapTestDefault.java:8:13: 'block' child has incorrect indentation level 12, expected level should be one of the following: 16, 20. [Indentation]
[ERROR] /var/tmp/HashMapTestDefault.java:9:9: 'block rcurly' has incorrect indentation level 8, expected level should be one of the following: 12, 16. [Indentation]
Audit done.
Checkstyle ends with 3 errors.
```

---

### Describe what you expect in detail

No violations should be reported. The code follows standard 4-space indentation:

- 4 spaces for method/annotation
- 8 spaces for variable declaration
- 12 spaces for statements inside instance initializer
- 8 spaces for closing braces

The double-brace initialization pattern `new HashMap<>() {{ ... }}` creates an anonymous class with an instance initializer. The indentation check incorrectly calculates the expected indentation for statements inside the instance initializer and for the closing braces. 
 _Originally posted by @SButterfly in [#2829](https://github.com/checkstyle/checkstyle/issues/2829#issuecomment-1387174388)_

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -2900,6 +2900,34 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
             getPath("InputIndentationAnonymousClassInMethodCurlyOnNewLine.java"), expected);
     }
 
+    // until #19415
+    @Test
+    public void testDoubleBraceInit() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+        checkConfig.addProperty("tabWidth", "4");
+        checkConfig.addProperty("basicOffset", "4");
+        checkConfig.addProperty("braceAdjustment", "0");
+        checkConfig.addProperty("caseIndent", "4");
+        checkConfig.addProperty("lineWrappingIndentation", "4");
+        checkConfig.addProperty("throwsIndent", "4");
+        checkConfig.addProperty("arrayInitIndent", "4");
+        final String[] expected = {
+            "27:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "block", 12, "16, 20"),
+            "28:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "block", 12, "16, 20"),
+            "29:9: " + getCheckMessage(MSG_ERROR_MULTI, "block rcurly", 8, "12, 16"),
+            "35:17: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "block", 16, "20, 24"),
+            "36:13: " + getCheckMessage(MSG_ERROR_MULTI, "block rcurly", 12, "16, 20"),
+            "42:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "block", 12, "16, 20"),
+            "43:9: " + getCheckMessage(MSG_ERROR_MULTI, "block rcurly", 8, "12, 16"),
+            "56:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "block", 12, "16, 20"),
+            "57:9: " + getCheckMessage(MSG_ERROR_MULTI, "block rcurly", 8, "12, 16"),
+            "63:17: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "block", 16, "20, 24"),
+            "64:17: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "block", 16, "20, 24"),
+            "65:13: " + getCheckMessage(MSG_ERROR_MULTI, "block rcurly", 12, "16, 20"),
+        };
+        verifyWarns(checkConfig, getPath("InputIndentationDoubleBraceInit.java"), expected);
+    }
+
     @Test
     public void testAnnotationDefinition() throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationDoubleBraceInit.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationDoubleBraceInit.java
@@ -1,0 +1,119 @@
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:0 exp:0
+
+import java.util.HashMap;                                               //indent:0 exp:0
+import java.util.HashSet;                                               //indent:0 exp:0
+import java.util.ArrayList;                                             //indent:0 exp:0
+import java.util.Map;                                                   //indent:0 exp:0
+
+/**                                                                     //indent:0 exp:0
+ * This test-input is intended to be checked using following            //indent:1 exp:1
+ * configuration:                                                       //indent:1 exp:1
+ *                                                                      //indent:1 exp:1
+ * arrayInitIndent = 4                                                  //indent:1 exp:1
+ * basicOffset = 4                                                      //indent:1 exp:1
+ * braceAdjustment = 0                                                  //indent:1 exp:1
+ * caseIndent = 4                                                       //indent:1 exp:1
+ * forceStrictCondition = false                                         //indent:1 exp:1
+ * lineWrappingIndentation = 4                                          //indent:1 exp:1
+ * tabWidth = 4                                                         //indent:1 exp:1
+ * throwsIndent = 4                                                     //indent:1 exp:1
+ */                                                                     //indent:1 exp:1
+public class InputIndentationDoubleBraceInit {                          //indent:0 exp:0
+
+    private final FluentCallChain configLog = new FluentCallChain();    //indent:4 exp:4
+    // behavior until #19415                                            //indent:4 exp:4
+    void test() {                                                       //indent:4 exp:4
+        var map = new HashMap<>() {{                                    //indent:8 exp:8
+            put("KEY1", "VALUE1");                                      //indent:12 exp:16,20 warn
+            put("KEY2", "VALUE2");                                      //indent:12 exp:16,20 warn
+        }};                                                             //indent:8 exp:12,16 warn
+    }                                                                   //indent:4 exp:4
+
+    void testNestedInIf() {                                             //indent:4 exp:4
+        if (true) {                                                     //indent:8 exp:8
+            var map = new HashMap<>() {{                                //indent:12 exp:12
+                put("KEY1", "VALUE1");                                  //indent:16 exp:20,24 warn
+            }};                                                         //indent:12 exp:16,20 warn
+        }                                                               //indent:8 exp:8
+    }                                                                   //indent:4 exp:4
+
+    Map<String, String> testReturnDoubleBrace() {                       //indent:4 exp:4
+        return new HashMap<>() {{                                       //indent:8 exp:8
+            put("KEY1", "VALUE1");                                      //indent:12 exp:16,20 warn
+        }};                                                             //indent:8 exp:12,16 warn
+    }                                                                   //indent:4 exp:4
+
+    void testSeparateLineInit() {                                       //indent:4 exp:4
+        var map = new HashMap<>() {                                     //indent:8 exp:8
+            {                                                           //indent:12 exp:12
+                put("KEY1", "VALUE1");                                  //indent:16 exp:16
+            }                                                           //indent:12 exp:12
+        };                                                              //indent:8 exp:8
+    }                                                                   //indent:4 exp:4
+
+    void testMethodArgDoubleBrace() {                                   //indent:4 exp:4
+        System.out.println(new HashMap<>() {{                           //indent:8 exp:8
+            put("KEY1", "VALUE1");                                      //indent:12 exp:16,20 warn
+        }});                                                            //indent:8 exp:12,16 warn
+    }                                                                   //indent:4 exp:4
+
+    void testWrappedMethodArgDoubleBrace() {                            //indent:4 exp:4
+        accept("first",                                                 //indent:8 exp:8
+            new HashMap<>() {{                                          //indent:12 exp:12
+                put("KEY1", "VALUE1");                                  //indent:16 exp:20,24 warn
+                put("KEY2", "VALUE2");                                  //indent:16 exp:20,24 warn
+            }});                                                        //indent:12 exp:16,20 warn
+    }                                                                   //indent:4 exp:4
+
+    void testNestedMethodArgDoubleBrace() {                             //indent:4 exp:4
+        accept(expectReadToEnd(new HashMap<>() {{                       //indent:8 exp:8
+                    put("KEY1", "VALUE1");                              //indent:20 exp:20
+                    put("KEY2", "VALUE2");                              //indent:20 exp:20
+                }}));                                                   //indent:16 exp:16
+    }                                                                   //indent:4 exp:4
+
+    void testChainedMethodArgDoubleBrace() {                            //indent:4 exp:4
+        foo()                                                           //indent:8 exp:8
+            .accept(new HashMap<>() {{                                  //indent:12 exp:12
+                        put("KEY1", "VALUE1");                          //indent:24 exp:24
+                        put("KEY2", "VALUE2");                          //indent:24 exp:24
+                    }})                                                 //indent:20 exp:20
+            .bar();                                                     //indent:12 exp:12
+    }                                                                   //indent:4 exp:4
+
+    void testKafkaStyleDoubleBrace() {                                  //indent:4 exp:4
+        foo()                                                           //indent:8 exp:8
+                .doAnswer(expectReadToEnd(new HashMap<>() {{            //indent:16 exp:16
+                            put("KEY1", "VALUE1");                      //indent:28 exp:28
+                            put("KEY2", "VALUE2");                      //indent:28 exp:28
+                        }}))                                            //indent:24 exp:24
+                .when(configLog).readToEnd();                           //indent:16 exp:16
+    }                                                                   //indent:4 exp:4
+
+    private void accept(String name, Map<String, String> values) {}     //indent:4 exp:4
+    private void accept(Object value) {}                                //indent:4 exp:4
+    private Map<String, String> expectReadToEnd(                        //indent:4 exp:4
+            Map<String, String> values) {                               //indent:12 exp:12
+        return values;                                                  //indent:8 exp:8
+    }                                                                   //indent:4 exp:4
+    private FluentCallChain foo() {                                     //indent:4 exp:4
+        return new FluentCallChain();                                   //indent:8 exp:8
+    }                                                                   //indent:4 exp:4
+
+    private static final class FluentCallChain {                        //indent:4 exp:4
+
+        private FluentCallChain accept(Map<String, String> values) {    //indent:8 exp:8
+            return this;                                                //indent:12 exp:12
+        }                                                               //indent:8 exp:8
+        private FluentCallChain bar() {                                 //indent:8 exp:8
+            return this;                                                //indent:12 exp:12
+        }                                                               //indent:8 exp:8
+        private FluentCallChain doAnswer(Object answer) {               //indent:8 exp:8
+            return this;                                                //indent:12 exp:12
+        }                                                               //indent:8 exp:8
+        private FluentCallChain when(FluentCallChain chain) {           //indent:8 exp:8
+            return this;                                                //indent:12 exp:12
+        }                                                               //indent:8 exp:8
+        private void readToEnd() {}                                     //indent:8 exp:8
+    }                                                                   //indent:4 exp:4
+}                                                                       //indent:0 exp:0

